### ISSUE TITLE
Type invariants ignored for local variables

### ISSUE THREAD

**Issue by SebastianS90** [Whiley/WhileyCompiler#626]
Local variables do not check type invariants. Consider this here:

```
type nat is (int x) where x >= 0

method m():
    nat x = -1
```

Compiles with verification!

The invariant is checked:
- On returning when the return type of a function/method has invariants
- On function/method call, when the parameter types have invariants
- But not for local variables

The language specification suggests that it should also be checked for local variables. (_holds for any instance of this type_)

**Comment by DavePearce** [comment]
Hey,

This is also a duplicate issue.  See #333 
I'll close the other one though, since you've done a better job writing it up :)

There are a bunch of known issues here, and it won't be resolved until I finished #563.

**Comment by DavePearce** [comment]
This is now *finally* fixed in the latest development version.  Phew!!!
